pkg/reconciler/apis/apiexportendpointslice/apiexportendpointslice_controller.go

```diff
@@ -353,11 +413,16 @@ func (c *controller) process(ctx context.Context, key string) error {
 	// If the object being reconciled changed as a result, update it.
 	oldResource := &Resource{ObjectMeta: old.ObjectMeta, Spec: &old.Spec, Status: &old.Status}
 	newResource := &Resource{ObjectMeta: obj.ObjectMeta, Spec: &obj.Spec, Status: &obj.Status}
-	if err := c.commit(ctx, oldResource, newResource); err != nil {
-		errs = append(errs, err)
+
+	// TODO: Maybe we can use different commiters? When is local or remote? Or we dont care on performance here yet?
+	if err := c.commitExternal(ctx, oldResource, newResource); err != nil {
+		if errors.IsConflict(err) {
+			logger.V(4).Info("requeueing APIExportEndpointSlice because of conflict", err)
+			return true, nil
+		}
```

Comment:
why does it matter? Different client?

Reply:
this was try to see why its failing. will remove. Conflicts where due to cache vs real object